docs/src/contribute/design/next-tedge-api/registration.md

```diff
@@ -0,0 +1,227 @@
+
+The following sequence diagram illustrates the automatic registration of a new child device via the first telemetry data published to the MQTT broker.
+
+```mermaid
+sequenceDiagram
+    ChildConnector->>MQTTBroker: Publish telemetry data to te/device/child01///m/example
+    MQTTBroker->>Mapper: Receive telemetry message
+    Mapper->>Mapper: Defer telemetry message (in memory storage)
+    Mapper->>Mapper: Infer entity/component from topic
+    Mapper ->> MQTTBroker: Publish registration message
+
+    MQTTBroker-->>Mapper: Receive registration message
+    Mapper->>Cloud: Register device
+    Cloud-->>Mapper: Device registered
+    Mapper->>Cloud: Publish deferred telemetry message
+
+    ChildConnector->>MQTTBroker: Publish telemetry data to te/device/child01///m/example
+    MQTTBroker->>Mapper: Receive telemetry message
+    alt Device/service is registered
+      Mapper->>Cloud: Publish telemetry message
+    end
+```
+
+### Inference of types by topic prefixes
+
+Automatic registration relies on entities and components publishing to declared namespaces which are associated with a specific namespace.
+
+The topic structure is used to infer the entity and component type by matching it against some pre-defined names. The pre-defined topic structure follows the following semantics:
+
+```text
+te/device/<device-id>/service/<name>
+```
+
+Where the `<device-id>` is either `main` (representing the thin-edge device), or a child device id
+
+```text
+te/device/[main|<other>]/service/<name>
+```
+
+Using the above semantics, the following table shows some examples showing the relationship between the topic name and the entity/component mapping.
+
+|Topic|Entity/Component Mapping|
+|--|--|
+|`te/device/main///`|Main device|
+|`te/device/main/service/nodered`|Service called `nodered` running on the main device|
+|`te/device/child01///`|Child device called `child01`. The device is an immediate child of the main device|
+|`te/device/fooBar///`|Child device called `fooBar`. The device is an immediate child of the main device|
+|`te/device/fooBar/service/nodered`|Service called `nodered` running on a child device called `fooBar`|
+
+
+:::tip
+You can apply your own semantic meaning to the topic segments, however it will require the devices/services to register themselves and the message needs to include what kind of entity/component they are, so that entity/component can be mapped correctly.
+
+See the [Advanced examples](#advanced-examples) for more details.
+:::
+
+## Examples
+
+### Register a device
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main' '{
+  "@type": "device",
+  "type": "Gateway"
+}'
+```
+
+Or use the a device can be registered using an explicit id:
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main' '{
+  "@type": "device",
+  "@id": "tedge001",
+  "type": "Gateway"
```

Comment:
What's the logic behind properties with or without an `@` prefix?

- What the difference between `@type` and `type`. I understand here that thin-edge uses the former but ignores the latter? But is this systematic?

Reply:
The idea was that property names starting with `@` relate to meta property which control how thin-edge treats the message, for example the `@type` tells thin-edge which entity type (device/service,child-device) should be used (as this can affect has the message is mapped to the IoT platform).

All other properties (not starting with `@`) are user-defined properties. It can be used by entities to describe the entity in the user's domain (rather than thin-edge's).